modules/integration/src/test/scala/scala/cli/integration/JmhTests.scala

```diff
@@ -1,106 +1,193 @@
 package scala.cli.integration
 
+import ch.epfl.scala.bsp4j as b
 import com.eed3si9n.expecty.Expecty.expect
 
 import java.nio.file.Files
 
+import scala.async.Async.{async, await}
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.jdk.CollectionConverters.*
 import scala.util.Properties
 
-class JmhTests extends ScalaCliSuite {
-  override def group: ScalaCliSuite.TestGroup = ScalaCliSuite.TestGroup.First
-
-  lazy val inputs: TestInputs = TestInputs(
-    os.rel / "benchmark.scala" ->
-      s"""package bench
-         |
-         |import java.util.concurrent.TimeUnit
-         |import org.openjdk.jmh.annotations._
-         |
-         |@BenchmarkMode(Array(Mode.AverageTime))
-         |@OutputTimeUnit(TimeUnit.NANOSECONDS)
-         |@Warmup(iterations = 1, time = 100, timeUnit = TimeUnit.MILLISECONDS)
-         |@Measurement(iterations = 10, time = 100, timeUnit = TimeUnit.MILLISECONDS)
-         |@Fork(0)
-         |class Benchmarks {
-         |
-         |  @Benchmark
-         |  def foo(): Unit = {
-         |    (1L to 10000000L).sum
-         |  }
-         |
-         |}
-         |""".stripMargin
-  )
-  lazy val expectedInOutput = """Result "bench.Benchmarks.foo":"""
-
-  test("simple") {
-    // TODO extract running benchmarks to a separate scope, or a separate sub-command
-    inputs.fromRoot { root =>
-      val res =
-        os.proc(TestUtil.cli, "--power", TestUtil.extraOptions, ".", "--jmh").call(cwd = root)
-      val output = res.out.trim()
-      expect(output.contains(expectedInOutput))
+class JmhTests extends ScalaCliSuite with JmhSuite with BspSuite {
+  override def group: ScalaCliSuite.TestGroup      = ScalaCliSuite.TestGroup.First
+  override protected val extraOptions: Seq[String] = TestUtil.extraOptions
+
+  for {
+    useDirective <- Seq(None, Some("//> using jmh"))
+    directiveString = useDirective.getOrElse("")
+    jmhOptions      = if (useDirective.isEmpty) Seq("--jmh") else Nil
+    testMessage = useDirective match {
+      case None            => jmhOptions.mkString(" ")
+      case Some(directive) => directive
+    }
+  } {
+    test(s"run ($testMessage)") {
```

Comment:
Maybe we should add tests for having both jmh and main class? I feel like if you have a main class and jhm tests it should run main by default but if you add `--jmh` explicitly it should rather run benchmarking tests.

Reply:
Yeah, so right now `--jmh` will run benchmarking tests.
I plan a future follow-up, where `--jmh` would either be extracted to a separate scope or a dedicated sub-command, but in the scope of this PR, `--jmh` forces benchmarking to be run, rather than the project's main class.
There's a bunch of TODOs in the `JmhTests.scala` source file that I plan to transform into issues as a follow-up.

Reply:
Okay, adding `//> using jmh`  will cause that too, but if you explicitly specify `--main-class` it will run the correct the chosen main so I think it is okay for now.